04week/mastermind.js

```diff
@@ -17,29 +21,141 @@ function printBoard() {
   }
 }
 
+//these are for testing 
 function generateSolution() {
   for (let i = 0; i < 4; i++) {
     const randomIndex = getRandomInt(0, letters.length);
     solution += letters[randomIndex];
   }
 }
 
+//this is a helper function for the generateSolution 
 function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+//this function should generate output that is the guess hints 
+  function generateHint(guessFormatted) {
   // your code here
+
+  // console.log('is my guess array coming through!', guessArray)
+  //converts my string to an array so I can loop through it 
+  let solutionArray = solution.split('')
+  console.log('this is what my solution looks like now ', solutionArray)
+  // console.log(`this is the length of my solution array :`,solutionArray.length)
+
+  //make your guess into an array
+  let guessArray = guessFormatted.split('')
+  console.log('guess array looks like now',guessArray)
+
+   //this variable will need to count how many letters are in the correct location 
+  let correctLetterLocations = 0  
+    // console.log('before the loop, my 100% accurate guesses are: ', correctLetterLocations)
+
+  //this will count how many letters match the solution 
+  let correctLetters = 0
+  // console.log('before the loop, the correct guesses letters are: ', correctLetters)
+
+  //first we're check if the location and letter matches 
+  let temporarySolution = [] 
+  let leftOvers = []
+
+  //using the index of the solution array, we want to check against each item in the index and verify if they match 
+  // console.log('this is the start loop for 100% accurate guesses')
+  for (let i = 0; i < solutionArray.length; i++) {
+    //if the letter and location matches
+    if(solutionArray[i] === guessArray[i]) {
+      // console.log(`YES ~ ${solutionArray[i]} matches ~ ${guessArray[i]}`)
+        //add its own array to count
+      temporarySolution.push(solutionArray[i])
+    } else {
+      // console.log(`NO ~ ${solutionArray[i]} does not matches ~ ${guessArray[i]}`)
+        //otherwise, put it in its own array to check against 
+      leftOvers.push(solutionArray[i])
+    }
+    correctLetterLocations = temporarySolution.length
+
+  }
+  // console.log(`temporary array : ${temporarySolution}`)
+  // console.log(`Left overs array? ~${leftOvers}`)
+  // console.log(`Guess array ${guessArray}`)
+
+
+  // console.log('AFTER 100% # guesses are ~' + correctLetterLocations)
+  console.log(`and the left overs? ~${leftOvers}`)
+  
+    //checks to see if any of the letters match 
+          //tally for duplicates
+
+  let once = []
+  let moreThan = []
+
+  guessArray.forEach((gLetter) => {
+    console.log(`what are my gLetters? ${gLetter}`)
+    // console.log(`can I see my left over array in here? ${leftOvers}`)
+    if((once.includes(gLetter)) || temporarySolution.includes(gLetter)) {     
```

Comment:
Not sure i understand the reasoning here, but the logic is not correct.
does not correctly handle a guess of "abcx", when the solution is "abcd"